### PR TITLE
Update curriculum based on feedback from review sessions

### PR DESCRIPTION
**Pull request by chrisdavidmills**
### Description

While reviewing the [Scrimba Frontend developer career path](https://scrimba.com/learn/frontend/) as a test, to see how much of our curriculum was covered in it, several update suggestions came to light. This PR makes those changes. This is mainly around removing some topics that were deemed unnecessary, and changing some descriptions.

Update: This PR now also includes several changes made as a result of discussing the curriculum with our UX design team. These include:

- Changing `1-standards-and-semantics` to `1-web-standards-and-html`, as the latter is thought to be more intuitive to developers. "Semantics" is a bit of an esoteric word, whereas HTML is crystal clear.
- Putting the note content in contextual locations, not all at the bottom of the modules. It was thought that this would make them more relevant and reach more people, and give us opportunities to make the UI more compact (e.g. including notes in expand/collapse boxes next to relevant learning outcomes).
- Removing unneeded `.md` files. There are quite a few to work with here, and we wanted to make the content as non-confusing as possible.
- Adding explainer notes to the top of the `README` and other non-content `.md` files. This has been done to help the design team figure out the purpose of each file, what should be used in the final publication, and what is relevant to the repo only.

### CODE REVIEW

**Review by couci — APPROVED** [review]
This looks good to me

**Comment by couci** [comment]
Adding Leo also for final review

**Comment by chrisdavidmills** [comment]
Hi @LeoMcA! I've finished making all the changes to the curriculum that have come out of reviewing Scrimba and talking to @AnujaRajput727 about the UX design work for the final publication. Can you please give it a quick review just to verify that I've not introduced any problems with my changes? Many thanks.

**Review by LeoMcA — APPROVED** [review]
two small things, otherwise looks good!

File: curriculum/README.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Introduction
 
+> Note: Main Introductory text for the curriculum, written to provide more depth on its purpose, target audience, scope, and usage. This information should be included somewhere in te final publication, perhaps as an FAQ or "About" page.
```
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply curriculum/README.md:3]: updated, ta.

File: curriculum/2-core/2-styling/3-10-debugging-css.md
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply curriculum/2-core/2-styling/3-10-debugging-css.md:1]: Good catch, thanks! I've updated this.

**Pull request merged**